h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java

```diff
@@ -599,15 +599,15 @@ private static int initializeFromChunks(Chunk[] chunks, int weight, DataInfo di,
 
 
             for (int j = 0; j < di._cats; ++j) {
-                if (!chunks[j].isNA(i)) {
+                if (chunks[j].atd(i) != 0) {
```

Comment:
Same issue as above - categorical and numerical values need to be treated differently.

Reply:
Remember to add the case when NA appears in chunks[j].atd(i).

Reply:
I do it with at8 and an if-else statement.